tools/ray_tpu/src/ray_tpu_controller.py

```diff
@@ -93,17 +93,14 @@ def __init__(
       tpu_name: str,
       startup_script: Optional[List[str]] = None,
       runtime_env: Optional[RayRuntimeEnv] = None,
-      head_addr: Optional[str] = None,
       **kwargs,
   ):
     if not ray.is_initialized():
       if runtime_env:
-        result = ray.init(runtime_env=dataclasses.asdict(runtime_env))
+        ray.init(runtime_env=dataclasses.asdict(runtime_env))
```

Comment:
could you try something like

`address = ray.init(address="auto")['address']`

Reply:
and test it with multislice example 

https://gist.github.com/yejingxin/4b7cb1199a7c3c33dbe96eb52a939814